modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/provider/zoom/ZoomPowwowServiceProvider.java

```diff
@@ -283,142 +288,63 @@ protected boolean deletePowwowMeeting(
 	}
 
 	protected void deleteZoomHost(PowwowServer powwowServer, String hostId) {
-		Map<String, String> params = new HashMap<>();
 
-		params.put("id", hostId);
+		List<String> resourceParams = new ArrayList<>(2);
+		resourceParams.add("users");
+		resourceParams.add(hostId);
 
-		JSONObject responseJSONObject = execute(
-			powwowServer, "user", "get", params);
+		JSONObject responseJSONObject = execute(powwowServer, resourceParams, Http.Method.GET, null);
 
 		String dept = responseJSONObject.getString("dept");
 
 		if (!dept.equals(_DEPT_API)) {
 			return;
 		}
 
-		responseJSONObject = execute(powwowServer, "user", "delete", params);
+		responseJSONObject = execute(powwowServer, resourceParams, Http.Method.DELETE, null);
 
-		String deletedAt = responseJSONObject.getString("deleted_at");
+		JSONObject errorJSONObject = responseJSONObject.getJSONObject("error");
 
-		if (deletedAt.equals(StringPool.BLANK)) {
+		if (errorJSONObject != null) {
 			throw new SystemException();
 		}
 	}
 
 	@Override
 	protected boolean endPowwowMeeting(
 		PowwowServer powwowServer, PowwowMeeting powwowMeeting) {
+		Map<String, Serializable> providerTypeMetadataMap = powwowMeeting.getProviderTypeMetadataMap();
 
-		JSONObject responseJSONObject = execute(
-			powwowServer, "meeting", "end", getParameterMap(powwowMeeting));
+		List<String> resourceParams = new ArrayList<>(3);
+		resourceParams.add("meetings");
+		resourceParams.add(String.valueOf(providerTypeMetadataMap.get("id")));
+		resourceParams.add("status");
+
+		JSONObject actionJSONObject = JSONFactoryUtil.createJSONObject();
+		actionJSONObject.put("action", "end");
 
-		String endedAt = responseJSONObject.getString("ended_at");
+		JSONObject responseJSONObject = execute(powwowServer, resourceParams, Http.Method.PUT, actionJSONObject);
+
+		JSONObject errorJSONObject = responseJSONObject.getJSONObject("error");
 
-		if (endedAt.equals(StringPool.BLANK)) {
+		if (errorJSONObject != null) {
 			return false;
 		}
 
 		return true;
 	}
 
 	protected JSONObject execute(
-		PowwowServer powwowServer, List<String> resourceParams, Http.Method method,
-		Map<String, String> parameterMap) {
```

Comment:
I'm pretty sure that we should keep param `Map<String, String> parameterMap`, because in some cases we'll need. 
For that param, maybe we need to consider the way to process params base on `method`, i'm not sure.
The new adding param should be called `jsonOptions` or `jsonParam`.